architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,259 @@
+
+   1. Communication:
+      1. Controller and UCPD:
+         1. Poll long-running operations (create, update, or delete) for a Recipe or a Deployment resource.
+         2. Create a Radius Resource Group if needed.
+         3. Create a Radius Application if needed.
+         4. Get a Radius resource like the Environment that is associated with the resource.
+         5. Create/Update/Delete a Recipe or a Deployment resource.
+         6. Create/Update/Delete a Secret for a Recipe or a Deployment resource.
+      2. Controller and the Kubernetes API Server:
+         1. Fetch a Recipe or a Deployment object.
+         2. Send events related to the operations running.
+         3. Update a Recipe or a Deployment object.
+         4. Create/Update/Delete a Secret associated with a Recipe or a Deployment object.
+         5. List Deployments by filtering them by a specific Recipe object.
+   2. Possible threats:
+      1. The controller component should have the minimum necessary Role-Based Access Control permissions to perform its tasks. It shouldn't have the permissions that it is not supposed to use.
+      2. A bad actor pretending to be UCPD, in case of compromised certificates, may lead to inconsistent resource data. The question is if a bad actor can create a process in the `radius-system` namespace that is going to pretend to be UCPD.
+      3. In case of compromised certificates of the Controller component, a bad actor can pretend to be the Controller and make requests to the UCPD and Kubernetes API Server.
+
+### Threats
+
+#### Threat: A bad actor pretending to be the Controller component to communicate with UCPD
+
+In case of compromised certificates and private key, an attacker can pretend to be the Controller component to communicate with UCPD.
+
+**Impact**: Controller Component and UCPD communicate in order to create/update/delete necessary resources. The impact of this threat would be unwanted operations on the resources.
+
+**Mitigation**:
+
+1. **mTLS**:
+
+   - **Description**: Ensure that mutual TLS is enabled for communication between the Controller and UCPD. This ensures that both parties authenticate each other, preventing unauthorized entities from impersonating the Controller.
+   - **Status**: Active
+
+2. **Strong Certificate Management**:
+
+   - **Description**: Implement strong certificate management practices, including regular rotation of certificates, using short-lived certificates, and monitoring for certificate anomalies. This reduces the risk of certificate compromise.
+   - **Status**:
+
+3. **Secure Storage**:
```

Comment:
For the mitigations that are already implemented, can we be more specific about how it is implemented? For example instead of "secrets management solutions", we should specify the specific secret manager used.